### ISSUE TITLE
Parameters for best model in paper

### ISSUE THREAD

**Issue by superphil0** [SinaHonari/RCN#5]
Hey,

I trained models now with the default parameters and got far worse results.
Are the default parameters the same as for the best model in your paper for which you report results?

**Comment by SinaHonari** [comment]
What numbers are you ending up with?

If you try the default parameters, you shouldn't get very different results. The results might slightly change by gpu, cudnn, or theano versions, but it should be within a small range.

Make sure, you do the preprocessing, data generation and data augmentation as mentioned in the Repo. Also try to plot how the model is converging over time and how the cost and error changes during training.

The parameters that are required to be tuned given the hardware/libraries are L2_coefficient and num_epochs. Some other hyper-parameters might also be helpful based on the model:
 --concat_pool_locations
--zero_non_pooled
--use_res_2
--weight_per_pixel (for sumNet model)

Optmization is also important. Check how your model is converging over time. I used Adadelta here and I got good results, but while training on some other models (not RCN) or on different cudnn/theano versions, I've got different results. If you see you get values that are too much off, you can replace it by Adam and you should get better results.

**Comment by superphil0** [comment]
Hi, I solely trained on 300W training data and tested on 300W test data. I do all preprocessing like mentioned in the README.
I train first the RCN and then the Denoising and run them with the 'plotting' and multi_probs flag on the test data. I extract for each file the points and do the reverse step of the preprocessing to get the points for the original image. I also run your evaluation script.

For the model which I trained (3K epochs the first and 100 epoch denoising) I get from your evaluation:
train set error is 0.0594494522707
valid set error is 0.061720666632
ibug set error is 0.109908486317
lfpw set error is 0.057029960346
Helen set error is 0.0570464965406

Which is in total for the test set (correctly weighted by number of images): 0.0593
In the paper you report: 0.0541

However I also make my own evaluation where I compare to other methods like MDM, there my evaluation (inter occular distance calculated by using the mean of 2 points of each eye like in your eval code) yields 0.084 for the test set

I veryfied my evaluation code by computing points for 
mdm (https://github.com/trigeorgis/mdm), 
3DDFA (http://www.cbsr.ia.ac.cn/users/xiangyuzhu/projects/3DDFA/main.htm)
CFSS (https://github.com/zhusz/CVPR15-CFSS)

I will check my code for extracting and computing the facepoints from RCN on the fullscale images again and will get back to you

**Comment by superphil0** [comment]
I checked the way I do the back transformation and also checked image outputs and could not find any mistakes, If you want I can send you the computed points and some examples

**Comment by SinaHonari** [comment]
Sorry for my late response.

I would recommend first checking the results you get for the RCN itself and then add the denoising model. Also, as I mentioned before, given the gpu, and cuda versions, the exact hyper-parameters might not get the same results, so you need to do a bit of hyper-parameter search. The results I got was with the best hyper-parameters on the hardware, cuda and theano settings and I did a comprehensive search.

**Comment by superphil0** [comment]
Hey, I could not reproduce the results even with your tips, I am sorry I don't have time at the moment to look into more detail. However, I want to thank you for publishing the code and the advice on training

**Comment by SinaHonari** [comment]
Hey there. Sorry to hear that. I will try to train a model and put it online and also report the parameters. It might take a bit since I'm busy with some other tasks.

**Comment by SinaHonari** [comment]
Hi,

I finally got some time to train a RCN model. The model is here:
https://github.com/SinaHonari/RCN/tree/master/trained_model/RCN_300W

Sorry for the delay!
